### PR TITLE
Fixed unnecessary user account lookup for device listing on user profile page.

### LINKED ISSUE #83: Cant update user if error in form with plugin activate
When we update a user in the admin (wp-admin/user-edit.php?user_id=1234) and there is a error on the form (example: username is empty and required) the url change to (wp-admin/user-edit.php) and your plugin trigger a die() in (class-devices.php -> custom_user_profile_fields)because there is no $_GET['user_id'] and prevent page to fully load. 
Why not using the $profileuser param for getting the user_id ?

`		// If is current user's profile (profile.php).
		if ( defined( 'IS_PROFILE_PAGE' ) && IS_PROFILE_PAGE ) {
			$user_id = get_current_user_id();
		} elseif( ! empty( $profileuser->ID ) ) {
			$user_id = absint( $profileuser->ID ); // phpcs:ignore
		} elseif ( ! empty( $_GET['user_id'] ) && is_numeric( $_GET['user_id'] ) ) { // phpcs:ignore
			// If is another user's profile page.
			$user_id = absint( $_GET['user_id'] ); // phpcs:ignore
		} else {
			// Otherwise something is wrong.
			die( 'No user id defined.' );
		}`

### PR DESCRIPTION
**Pull request by pesseba**
Hi @contactjavas. It is a bugfix for the issue #83

fixes #83

### CODE REVIEW

**Review by contactjavas — APPROVED** [review]
thanks @pesseba ! Seems good to me :)

**Review by dominic-ks — APPROVED** [review]
Yep I tested this and it fixes the issue. Not sure if you guys have seen my notes in Discord but I added [this new branch](https://github.com/usefulteam/jwt-auth/tree/wp-org-master) to get the plugin back on wp.org without merging the refresh token changes in there with no warnings to users, so this pr won't fix that for people who have the plugin from there at the moment...

**Review by sun — APPROVED** [review]

File: class-devices.php [hunk 1]
```diff
@@ -311,21 +311,10 @@ public function remove_device() {
 	 */
 	public function custom_user_profile_fields( $profileuser ) {
```
Comment (sun) [inline class-devices.php:312]: https://developer.wordpress.org/reference/hooks/edit_user_profile/
```suggestion
	public function custom_user_profile_fields( $profile_user ) {
```

File: class-devices.php [hunk 2]
```diff
@@ -311,21 +311,10 @@ public function remove_device() {
...
-		} else {
-			// Otherwise something is wrong.
-			die( 'No user id defined.' );
-		}
-
 		?>
 		<h2><?php echo __( 'Connected Devices', 'jwt-auth' ); ?></h2>
 		<div id="jwt_auth_devices" style="width:33%">
-			<?php echo do_shortcode( '[jwt_auth_devices user_id=' . $user_id . ']' ); ?>
+			<?php echo do_shortcode( '[jwt_auth_devices user_id=' . $profileuser->ID . ']' ); ?>
```
Comment (sun) [inline class-devices.php:317]:
```suggestion
			<?php echo do_shortcode( '[jwt_auth_devices user_id=' . $profile_user->ID . ']' ); ?>
```

**Comment by sun** [comment]
@pesseba Any feedback on the suggested changes?  Or shall we just apply them and move forward?

**Pull request merged**